src/collective/easyform/actions.py

```diff
@@ -349,12 +353,27 @@ def get_header_info(self, fields, request, context,
     def serialize(self, field):
         """Serializa field to save to XML.
         """
-        if isinstance(field, set) or isinstance(field, list):
+        if isinstance(field, (set, list, tuple)):
             list_value = list([str(f) for f in field])
             return dumps(list_value)
+        if isinstance(field, dict):
+            dict_value = {str(key): str(val) for key, val in field.items()}
+            return dumps(dict_value)
         if isinstance(field, RichTextValue):
             return field.raw
-        return safe_unicode(field)
+        if isinstance(field, datetime):
+            return field.strftime("%Y/%m/%d, %H:%M:%S")
+        if isinstance(field, date):
+            return field.strftime("%Y/%m/%d")
+        if isinstance(field, timedelta):
+            return str(field)
+        if isinstance(field, (int, float, Decimal, bool)):
+            return str(field)
+        if isinstance(field, six.string_types):
+            return safe_unicode(field)
+        if  isinstance(field, six.text_type):
+            field
```

Comment:
missing `return` but you never get here anyways when you check for `six.string_types` before ... this is superfluos ... simply remove it.
If you want some kind of "super_safe_unicode" in the end then there's `six.binary_type` check missing ... 
or do I get something wrong here?

Reply:
@petschki , all other type (including binary) will be returned by our last statement, return safe_unicode(repr(field))